rclcpp/include/rclcpp/node.hpp

```diff
@@ -64,6 +64,17 @@
 namespace rclcpp
 {
 
+/// Used as argument in create_publisher and create_subscriber.
+enum class IntraprocessSetting
+{
+  /// Explicitly enable intraprocess comm at publisher/subscription level.
+  ENABLE,
```

Comment:
Let's not use ALL_CAPS, as that is often used by macros, and could collide here. (imagine someone did `#define ENABLE ...`, which is not unthinkable.

Instead, let's do `Enable`, `Disable`, and `Default` (or `DefaultFromNode` but that doesn't look very nice to me, another option is maybe `NodeDefault`).

Reply:
Up to now, most of the "enum class" in rclcpp follow the ALL_CAPS style, except of `CallbackGroupType`. I agree we should only use it in macros, so I will change it.

Reply:
Yeah, there might be other cases that should change, but let's arc towards what we think is right.